crates/rover-client/src/error.rs

```diff
@@ -140,6 +139,11 @@ pub enum RoverClientError {
     #[error("The variant `{graph_ref}` is a non-contract variant. This operation is only possible for contract variants.")]
     ExpectedContractVariant { graph_ref: GraphRef },
 
+    /// This error occurs when a request returns data as though it was both a contract and non-contract variant.
+    /// This usually indicates an error from studio, but is very unlikely to occur in practice
+    #[error("The graph `{graph_ref}` has returned a response indicating it is both a contract and non-contract variant.")]
+    ContractAndNonContractVariant { graph_ref: GraphRef },
```

Comment:
__NIT__: should we default to the non-contract variant data when that's the case?

Reply:
Yer that sounds sensible, I've rejigged the matches on the returns from the query and added a lot more comments about why we're doing that